### ISSUE TITLE
Uphold `/capabilities` endpoint spec changed — need to adjust `get_capabilities`

### ISSUE THREAD

**Issue by szilardszaloki** [brave/brave-browser#23849]
## Description 

Due to a specification change of Uphold's `/capabilities` endpoint, users that are required to provide information to Uphold (e.g. CDD, EDD, etc.) are now allowed to verify with Uphold, although they cannot send (tip), and/or receive (monthly payouts).

## Miscellaneous Information:

cc: @Miyayes @brave/rewards-client

**Comment by Miyayes** [comment]
Does this need a separate Android implementation?

**Comment by szilardszaloki** [comment]
No, the desktop change will take care of it on Android, too.

**Comment by kjozwiak** [comment]
The above will require `1.42.67` or higher for `1.42.x` verification 👍

**Comment by GeetaSarvadnya** [comment]
Verification is in-progress

```
Brave | 1.42.65 Chromium: 103.0.5060.114 (Official Build) beta (64-bit)
-- | --
Revision | a1c2360c5b02a6d4d6ab33796ad8a268a6128226-refs/branch-heads/5060@{#1124}
OS | Windows 10 Version 21H2 (Build 19044.1826)

```
- Verified the test plan from https://github.com/brave/brave-core/pull/14057#issuecomment-1183539618

**Comment by LaurenWags** [comment]
For Android, I'd recommend testing Case 1, 2, 4 from https://github.com/brave/brave-browser/issues/23849#issuecomment-1185698941. Spot checking 3, 5, 6, 7 if there's time will be sufficient.

**Comment by Uni-verse** [comment]
Verification in progress on Samsung Galaxy Tab S7 using

Brave	1.42.81 Chromium: 104.0.5112.69 (Official Build) (64-bit) 
Revision	7ce2902023c722af8564068e6b26e934b83fd774-refs/branch-heads/5112@{#1213}
OS	Android 12; Build/SP1A.210812.016

**Comment by srirambv** [comment]
Verification in progress on Samsung Galaxy S21 using

Brave	1.42.81 Chromium: 104.0.5112.69 (Official Build) (64-bit) 
Revision	7ce2902023c722af8564068e6b26e934b83fd774-refs/branch-heads/5112@{#1213}
OS	Android 12; Build/SP1A.210812.016

Note - while running the below tests you may encounter known issue https://github.com/brave/brave-browser/issues/16376 on the brave://rewards-internals page.

<details>
<summary>Case 1 - Sufficient capabilities - PASSED</summary>

**Scenario 1 - Clean profile - PASSED**
1. Launched with a clean profile and staging env
2. Enabled Rewards and verify w/ Uphold account that has sufficient capabilities
3. Confirmed `/v0/me/capabilities` returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```
:method: GET
:authority: api-sandbox.uphold.com
:scheme: https
:path: /v0/me/capabilities
sec-fetch-site: none
sec-fetch-mode: no-cors
sec-fetch-dest: empty
user-agent: Mozilla/5.0 (Linux; Android 12) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.5112.69 Safari/537.36
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9,es-ES;q=0.8,es;q=0.7,ru-RU;q=0.6,ru;q=0.5,de-DE;q=0.4,de-AT;q=0.3,de;q=0.2,fr-FR;q=0.1,fr;q=0.1,es-US;q=0.1,ja-JP;q=0.1,ja;q=0.1  {
	"category": "permissions",
	"enabled": true,
	"key": "receives",
	"name": "Receives",
	"requirements": [],
	"restrictions": []
}, {
	"category": "permissions",
	"enabled": true,
	"key": "sends",
	"name": "Sends",
	"requirements": [],
	"restrictions": []
},
```

</details>

4. Confirmed balance displayed on brave://rewards and panel
5. Confirmed no notifications/messages about needing to complete additional requirements

Example | Example
-- | --
![screenshot-1659137020982](https://user-images.githubusercontent.com/17885425/181861881-9ab46218-f712-4503-8bcf-44759360d986.png) | ![screenshot-1659137052444](https://user-images.githubusercontent.com/17885425/181861882-b6049fd1-8cb6-4479-bfd0-b30441f65a5e.png)


**Scenario 2 - Upgrade profile - on brave://rewards page prior to Upgrade - PASSED**

1. Launched 1.41.x with a clean profile and staging env
2. Enabled Rewards and verify w/ Uphold account that has sufficient capabilities
3. Confirmed `/v0/me/capabilities` returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380
4. Confirmed balance displayed on brave://rewards and panel
5. Ensured brave://rewards page was active tab
6. Closed 1.41.x and "Upgraded" version (renamed profile)
7. Launched with upgraded version
8. Confirmed `/v0/me/capabilities` still returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```
:method: GET
:authority: api-sandbox.uphold.com
:scheme: https
:path: /v0/me/capabilities
sec-fetch-site: none
sec-fetch-mode: no-cors
sec-fetch-dest: empty
user-agent: Mozilla/5.0 (Linux; Android 12) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.5112.69 Mobile Safari/537.36
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9,es-MX;q=0.8,es;q=0.7,de-AT;q=0.6,de;q=0.5,pt-BR;q=0.4,pt;q=0.3,de-DE;q=0.2,en-GB;q=0.1,ru-RU;q=0.1,ru;q=0.1,es-US;q=0.1,ja-JP;q=0.1,ja;q=0.1,pt-PT;q=0.1,en-CA;q=0.1,fr-FR;q=0.1,fr;q=0.1,en-DE;q=0.1

 {
	"category": "permissions",
	"enabled": true,
	"key": "receives",
	"name": "Receives",
	"requirements": [],
	"restrictions": []
}, {
	"category": "permissions",
	"enabled": true,
	"key": "sends",
	"name": "Sends",
	"requirements": [],
	"restrictions": []
}, 
............
```

</details>

9. Confirmed balance displayed on brave://rewards and panel
10. Confirmed no notifications/messages about needing to complete additional requirements

**Scenario 3 - Upgrade profile - brave://rewards page is not open prior to Upgrade - PASSED**

1. Launched 1.41.x with a clean profile and staging env
2. Enabled Rewards and verify w/ Uphold account that has sufficient capabilities
3. Confirmed `/v0/me/capabilities` returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380
4. Confirmed balance displayed on brave://rewards and panel
5. Ensured brave://rewards page was not open
6. Closed 1.41.x and "Upgraded" version (renamed profile)
7. Launched with upgraded version
8. Confirmed `/v0/me/capabilities` still returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```
:method: GET
:authority: api-sandbox.uphold.com
:scheme: https
:path: /v0/me/capabilities
sec-fetch-site: none
sec-fetch-mode: no-cors
sec-fetch-dest: empty
user-agent: Mozilla/5.0 (Linux; Android 12) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.5112.69 Mobile Safari/537.36
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9,es-MX;q=0.8,es;q=0.7,de-AT;q=0.6,de;q=0.5,pt-BR;q=0.4,pt;q=0.3,de-DE;q=0.2,en-GB;q=0.1,ru-RU;q=0.1,ru;q=0.1,es-US;q=0.1,ja-JP;q=0.1,ja;q=0.1,pt-PT;q=0.1,en-CA;q=0.1,fr-FR;q=0.1,fr;q=0.1,en-DE;q=0.1

 {
	"category": "permissions",
	"enabled": true,
	"key": "receives",
	"name": "Receives",
	"requirements": [],
	"restrictions": []
}, {
	"category": "permissions",
	"enabled": true,
	"key": "sends",
	"name": "Sends",
	"requirements": [],
	"restrictions": []
} 
............
```

</details>

9. Confirmed balance displayed on panel
10. Confirmed no notifications/messages about needing to complete additional requirements

</details>

<details>
<summary>Case 2 - Insufficient capabilities, receives=1 with requirements - IN PROGRESS</summary>



</details>

**Comment by GeetaSarvadnya** [comment]
Verification PASSED on `Samsung Galaxy Tab A7 version 11` running `Bravemonoarm64.apk_1.42.85`

<details>
<summary>Case 3 - Insufficient capabilities, receives=2 - PASSED</summary>

**Scenario 1 - Clean profile - PASSED**

1. Configured a Charles Proxy rule for the body of `/v0/me/capabilities` as per https://github.com/brave/brave-core/pull/14057#issue-1294849380, ensure rule is running
2. Launched with a clean profile and staging env
3. Enabled Rewards and linked Uphold account
4. Confirmed `/v0/me/capabilities` was overwritten per the Charles Proxy rule as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```


```

</details>

5. Confirmed message displayed on brave://rewards page 
6. Confirmed no notification on BAT logo in URL bar
7. Confirmed balance not shown on brave://rewards page or panel
8. Confirmed status was as expected on brave://rewards-internals

Example | Example
-- | --



**Scenario 2 - Upgrade profile - on brave://rewards page prior to Upgrade - PASSED**

1. Using same Charles Proxy rule from "Clean Profile" case for this scenario, launched 1.41.x with a clean profile and staging env
2. Enabled Rewards and verify w/ Uphold account 
3. Confirmed `/v0/me/capabilities` returned as expected for this case per Charles Proxy rule written as per https://github.com/brave/brave-core/pull/14057#issue-1294849380
4. Confirm unable to verify wallet due to "sends" or "receives" being false (implemented with previously with https://github.com/brave/brave-browser/issues/22508) 
- modal with message is shown 
- no balance displayed on rewards page or panel
5. Ensured brave://rewards page was active tab
6. Closed 1.41.x 
7. "Upgraded" version (renamed profile) and Launched with upgraded version
8. Confirmed status was as expected on brave://rewards-internals (still `NOT CONNECTED`)
9. Attempt to link account again (with Charles Proxy still running the rewrite rule)
10. Confirmed `/v0/me/capabilities` still returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```

 

```

</details>

11. Confirmed modal message is shown
12. Confirmed balance not displayed on rewards page or panel
13. Confirmed status was as expected on brave://rewards-internals (still `NOT CONNECTED`)

Example | Example
-- | -- 



**Scenario 3 - Upgrade profile - brave://rewards page not open prior to Upgrade - PASSED**

1. Using same Charles Proxy rule from "Clean Profile" case for this scenario, launched 1.41.x with a clean profile and staging env
2. Enabled Rewards and verify w/ Uphold account 
3. Confirmed `/v0/me/capabilities` returned as expected for this case per Charles Proxy rule written as per https://github.com/brave/brave-core/pull/14057#issue-1294849380
4. Confirm unable to verify wallet due to "sends" or "receives" being false (implemented with previously with https://github.com/brave/brave-browser/issues/22508) 
- modal with message is shown 
- no balance displayed on rewards page or panel
5. Ensured brave://rewards page was not open
6. Closed 1.41.x 
7. "Upgraded" version (renamed profile) and Launched with upgraded version
8. Confirmed status was as expected on brave://rewards-internals (still `NOT CONNECTED`)
9. Attempt to link account again (with Charles Proxy still running the rewrite rule)
10. Confirmed `/v0/me/capabilities` still returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```

```

</details>

11. Confirmed modal message is shown
12. Confirmed balance not displayed on rewards page or panel
13. Confirmed status was as expected on brave://rewards-internals (still `NOT CONNECTED`)

Example | Example
-- | -- 



</details>

<details>
<summary>Case 4 - Insufficient capabilities, sends=1 with requirements - PASSED</summary>

**Scenario 1 - Clean profile - PASSED**

1. Configured a Charles Proxy rule for the body of `/v0/me/capabilities` as per https://github.com/brave/brave-core/pull/14057#issue-1294849380, ensure rule is running
2. Launched with a clean profile and staging env
3. Enabled Rewards and linked Uphold account
4. Confirmed `/v0/me/capabilities` was overwritten per the Charles Proxy rule as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```

```

</details>

5. Confirmed message displayed on brave://rewards page 
6. Confirmed no notification on BAT logo in URL bar
7. Confirmed balance not shown on brave://rewards page or panel
8. Confirmed status was as expected on brave://rewards-internals

Example | Example
-- | --




**Scenario 2 - Upgrade profile - on brave://rewards page prior to Upgrade - PASSED**

1. Using same Charles Proxy rule from "Clean Profile" case for this scenario, launched 1.41.x with a clean profile and staging env
2. Enabled Rewards and verify w/ Uphold account 
3. Confirmed balance displayed on brave://rewards and panel
4. Confirmed `/v0/me/capabilities` returned as expected for this case per Charles Proxy rule written as per https://github.com/brave/brave-core/pull/14057#issue-1294849380
5. Ensured brave://rewards page was active tab
6. Closed 1.41.x 
7. "Upgraded" version (renamed profile) and Launched with upgraded version
8. Confirmed `/v0/me/capabilities` still returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```

```

</details>

10. Confirmed modal and notification (on BAT logo) are shown
11. Confirmed balance no longer displayed on rewards page or panel
12. Confirmed status was as expected on brave://rewards-internals (`VERIFIED` --> `NOT CONNECTED`)
13. Confirmed logs showed the below as per the PR:

```


```

Example | Example | Example
-- | -- | --



**Scenario 3 - Upgrade profile - brave://rewards page is not open prior to Upgrade - PASSED**

1. Using same Charles Proxy rule from "Clean Profile" case for this scenario, launched 1.41.x with a clean profile and staging env
2. Enabled Rewards and verify w/ Uphold account 
3. Confirmed balance displayed on brave://rewards and panel
4. Confirmed `/v0/me/capabilities` returned as expected for this case per Charles Proxy rule written as per https://github.com/brave/brave-core/pull/14057#issue-1294849380
5. Ensured brave://rewards page was not open tab
6. Closed 1.41.x 
7. "Upgraded" version (renamed profile) and Launched with upgraded version
8. Confirmed `/v0/me/capabilities` still returned as expected for this case per https://github.com/brave/brave-core/pull/14057#issue-1294849380

<details>

```
```

</details>

10. Confirmed only notification (on BAT logo) is shown (no modal)
11. Confirmed balance no longer displayed on rewards page or panel
12. Confirmed status was as expected on brave://rewards-internals (`VERIFIED` --> `NOT CONNECTED`)
13. Confirmed logs showed the below as per the PR:

```
```

Example | Example
-- | -- 

</details>
